src/main/kotlin/com/sourcegraph/cody/agent/protocol/Range.kt

```diff
@@ -12,4 +14,15 @@ data class Range(val start: Position, val end: Position) {
   fun toSearchRange(): RangePair = RangePair(start.line.plus(1), end.line)
 
   fun length() = end.line - start.line + 1
+
+  fun toOffsets(document: Document): Pair<Int, Int> {
+    return Pair(start.toOffset(document), end.toOffset(document))
+  }
+
+  // Return true if the agent gave us all zeroes for the range.
+  fun isZero() = start.isZero() && end.isZero()
+
+  companion object {
+    fun nullRange() = Range(Position(0, 0), Position(0, 0))
```

Comment:
Mildly nervous about this "null range", "zero position", etc. idea. Because if you start a new file and tell Cody to do stuff, the position is going to be like this.

Could we make the protocol actually use undefined or null for missing, and trust when zeros do appear that we meant it?

Reply:
Apologies; there were 3 unused methods here and this was one. I've done a sweep for other warnings as well.